p2p/security/tls/transport.go

```diff
@@ -111,11 +128,22 @@ func (t *Transport) setupConn(tlsConn *tls.Conn, remotePubKey ci.PubKey) (sec.Se
 	if err != nil {
 		return nil, err
 	}
+
+	nextProto := tlsConn.ConnectionState().NegotiatedProtocol
+	// The special nextProto ID "libp2p" was used in previous versions upto
+	// v0.23.2 as the TLS ALPN extension field. If we see this special ID
+	// selected, that means we are handshaking with an old version of libp2p.
```

Comment:
Slight rewording, as there are other libp2p implementations than go-libp2p out there.
```suggestion
	// The special ALPN value "libp2p" is used by libp2p implementations that
	// don't support early muxer negotiation.
```

Reply:
good point to cover other implementations too. done.